From a3ae70506748b86a9312ebac285ed04f91a0eb24 Mon Sep 17 00:00:00 2001
From: Peter Geis <pgwipeout@gmail.com>
Date: Mon, 21 Feb 2022 20:31:24 -0500
Subject: [PATCH 1/5] spl: support adc drivers in spl

In order to handle the rockchip recovery handler in spl, we need the adc
code to be available in spl.
Add a toggle to allow adc drivers to function in spl.

Signed-off-by: Peter Geis <pgwipeout@gmail.com>
---
 common/spl/Kconfig | 5 +++++
 drivers/Makefile   | 1 +
 2 files changed, 6 insertions(+)

diff --git a/common/spl/Kconfig b/common/spl/Kconfig
index 2c042ad3..b77187eb 100644
--- a/common/spl/Kconfig
+++ b/common/spl/Kconfig
@@ -587,6 +587,11 @@ config SPL_FIT_IMAGE_TINY
 	  ensure this information is available to the next image
 	  invoked).
 
+config SPL_ADC
+	bool "Support ADC drivers in SPL"
+	help
+	  Enable ADC drivers in SPL.
+
 config SPL_CACHE
 	bool "Support CACHE drivers"
 	help
diff --git a/drivers/Makefile b/drivers/Makefile
index 29be78a3..86a584e2 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -1,5 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0+
 
+obj-$(CONFIG_$(SPL_)ADC) += adc/
 obj-$(CONFIG_$(SPL_TPL_)BLK) += block/
 obj-$(CONFIG_$(SPL_TPL_)BOOTCOUNT_LIMIT) += bootcount/
 obj-$(CONFIG_$(SPL_TPL_)BUTTON) += button/
-- 
2.34.8

